crates/nargo/src/ops/test.rs

```diff
@@ -25,44 +27,108 @@ pub fn run_test<B: BlackBoxFunctionSolver>(
             // otherwise constraints involving these expressions will not error.
             let circuit_execution =
                 execute_circuit(blackbox_solver, program.circuit, WitnessMap::new(), show_output);
+            test_status_program_compile_pass(test_function, circuit_execution)
```

Comment:
```suggestion
            test_status_program_execution_pass(test_function, circuit_execution)
```

Reply:
This should be compile_pass, so we only call it if compilation has passed. execution passing is checked in compile_pass. 

Its done this way because we evaluate parts of the circuit at compile time, its one of the reasons why this code is kind of complicated

Reply:
Ah yes I see we are actually checking execution pass/fail inside of the compile_pass